Improve accuracy of swagger output

rygwdn opened the pull request:
- Allow omitting endpoints on the viewset
- Improve default status codes
- DELETE responses are empty
- add hooks to augment operations
- add support for pagination
- remove trailing "Serializer" from definitions

rygwdn committed:
Remove check_yaml_methods.

This causes issues with ViewSets, since the viewset is actually created
with multiple action mappings for the different url patterns.

rygwdn committed:
Use correct status codes for POST and DELETE.

These match up to the default codes sent by DRF.

rygwdn committed:
Do not include definitions for excluded endpoints.

rygwdn committed:
Do not include paths that have no operations.

rygwdn committed:
Remove schema from DELETE operations.

- Extracted new method to make the code easier to follow.
- Added ability for methods to override the default responses in their
  docstring.

rygwdn committed:
Add hooks to update operations based on callback and method.

rygwdn committed:
Add support for pagination definitions.

- Specified a default page definition which gets templated
- Allow specifying different default definition, and overriding per
  operation
- Default to using the page definition for "list" operations

rygwdn committed:
Remove trailing "Serializer" from definition names.

rygwdn committed:
Allow forcing pagination (or not) through yaml.

npars commented on rest_framework_swagger/docgenerator.py:
```diff
@@ -139,29 +151,68 @@ def get_operations(self, api_endpoint, introspector):
-                }
-            }
+            # Remove blank response objects - allows yaml to remove default responses
+            for code in response_messages:
```
Sketchy modification of collection while iterating on it, could explicitly make a copy of keys before iterating or do a comprehension and create a new dict

npars commented on rest_framework_swagger/docgenerator.py:
```diff
@@ -139,29 +151,68 @@ def get_operations(self, api_endpoint, introspector):
+    def _get_default_response_object(self, operation_method, response_type):
+        schema = {
+            '$ref': '#/definitions/' + response_type
+        } if response_type != 'object' else {
```
Not sold on this ternary being cleaner than an ordinary if/else

rygwdn committed:
Clean up page templating.

- Recurse through the dict instead of using json and string replace
- Clean up some other code

npars commented:
:+1:

mdentremont commented:
+2
